chemked/validation.py

```diff
@@ -47,6 +49,49 @@
                   'time': 'second',
                   }
 
+def compare_name(given_name, family_name, question_name):
+    """Compares a name in question to a specified name separated into given and family.
+
+    The name in question ``question_name`` can be of varying format, including
+    "Kyle E. Niemeyer", "Kyle Niemeyer", "K. E. Niemeyer", "KE Niemeyer", and
+    "K Niemeyer". Other possibilities include names with hyphens such as
+    "Chih-Jen Sung", "C. J. Sung", "C-J Sung".
+    """
+    # lowercase everything
+    given_name = given_name.lower()
+    family_name = family_name.lower()
+    question_name = question_name.lower()
+
+    # split name in question by , <space> - . ! ? :
+    name_split = list(filter(None, re.split("[, \-.!?:]+", question_name)))
+    first_name = [name_split[0]]
```

Comment:
Well now you're just being difficult 😿

Right now it assumes the names are written in the correct order. I could have it look for a comma, and consider anything with a comma written as above, as in `'Niemeyer, Kyle E'` -> `['Niemeyer', 'Kyle', 'E']` -> `['Kyle', 'E', 'Niemeyer']` and then handled normally. This seems pretty easy, so I'll do this.

Since the period is at the end of that string, it would just be removed.

*or* we could just require names formatted correctly 😀